### DISCUSSION TITLE
hooks是对应操作之前执行哈。如果想之后执行呢？

### DISCUSSION THREAD

**Discussion by Anne3720** [antvis/XFlow#365] [Q&A]
比如我想在addNode、delNode、updateNode更新之后，把图的数据直接存到后台去。可以用hooks实现吗？

**Comment by RRDAWLX** [comment]
[这篇文档](https://xflow.antv.vision/api/hooks/)的示例中在 addNode 钩子的回调中给 args.createNodeService 赋值。
[另一篇文档](https://xflow.antv.vision/api/commands/node-add)中说“在执行 AddNodeCommand 时会先执行 ICreateNodeService”。
所以推测 hooks 是在对应的操作之前执行的。
